patches/tModLoader/Terraria/ModLoader/ModBiome.cs

```diff
@@ -44,7 +44,7 @@ public abstract class ModBiome : ModSceneEffect, IShoppingBiome, ILocalizedModTy
 
 	public GameContent.Bestiary.ModBiomeBestiaryInfoElement ModBiomeBestiaryInfoElement { get; internal set; }
 
-	string IShoppingBiome.NameKey => Name;
+	string IShoppingBiome.NameKey => $"Mods.{Mod.Name}.TownNPCMoodBiomes.{Name}";
```

Comment:
Is there a reason for this to be different to `DisplayName`?

I looked up references to `NameKey` and they're pretty scares, it seems like it would be better to replace it with `LocalizedText DisplayName` (on `IShoppingBiome`) and pass the `LocalizedText` everywhere.

It is only ever used to create the localization substitutions object via 
`new { BiomeName = BiomeNameByKey(...)`

but we could just do `new { BiomeName = DisplayName }` since `LocalizedText` instances work fine as substitution args

Reply:
Alright, but same feedback here as for NPC mood in general. Better to use `GetLocalizationKey("TownNPCDialogueName")`